bftengine/src/bftengine/KeyManager.cpp

```diff
@@ -76,7 +77,7 @@ std::string KeyManager::generateCid() {
 std::string KeyManager::onKeyExchange(KeyExchangeMsg& kemsg, const uint64_t& sn) {
   if (kemsg.op == KeyExchangeMsg::HAS_KEYS) {
     LOG_INFO(KEY_EX_LOG, "Has key query arrived, returning " << std::boolalpha << keysExchanged << std::noboolalpha);
-    if (!keysExchanged) return std::string(KeyExchangeMsg::hasKeysFalseReply);
+    if (!keysExchanged && enableHasKeysReq_) return std::string(KeyExchangeMsg::hasKeysFalseReply);
```

Comment:
I'm very confused by the variable name. How about we keep the same name that we use in the configuration?
```
if (!keysExchange && keyExchangeOnStart)
```

Reply:
sure